src/Core/Enum.php

```diff
@@ -56,10 +57,11 @@ public function __construct($value)
      *
      * @param string $value
      * @return bool the enum has the value
+     * @throws \ReflectionException
```

Comment:
Instead of propagating `@throws \ReflectionException` up, how about removing it from `toArray()` phpDoc (since `static::class` is guaranteed to exist)?